Implemented the Club Registration Screen (UI + Logic)

nirvan73 opened the pull request:
## 📝Overview
- This PR fixes or fixes part of https://github.com/bsoc-bitbyte/Nudj/issues/50
- This PR does the following: Implements the club registration screen with its (UI + ViewModel) based on the created and verified Figma design , while keeping the colors , theme , fonts and shapes in mind using the User Repository.

## ✅ Checklist
- ✅ My code follows the project's style and structure
- ✅ Matches Figma design specifications
- ✅ Tested locally with no errors

## 📸 Proof that changes are correct
- Added screen recording of the my clubs registration screen layout for any UI changes/feedback.
- Also the previews for the composable functions implemented inside the ClubRegistrationScreen1.kt and ClubRegistrationScreen2.kt files are made inside the file itself.

https://github.com/user-attachments/assets/e3950273-a41b-4e27-aec8-6bd76c3b6f70

## 🗒️ Additional Notes
- Figma mockup link: [Here](https://www.figma.com/design/Zts7XEdu6oIpMYaTQ8yb7i/Nudj-Contributor-Designs?node-id=0-1&p=f&t=yr2twmvpglNHd7pr-0)
- Added the required images in drawable file.
- Registered the ClubRegistrationScreen in the MainActivity.kt .
- Added the circular progress indicator to ensure a smooth and seamless user experience.
- Used the BaseViewModel and UI states for state management.

hemang-mishra commented:
@nirvan73 Better remove club email text field. It's not required as we need the clubs to login through their official mail only to prevent multiple accounts by the same club name.

hemang-mishra commented:
Really great work btw!! Keep going.

nirvan73 commented:
Updated the files as per the requested changes and removed the Club Email input section. Now, the email of the currently logged-in user is used instead.

hemang-mishra commented:
Lgtm
Great work!
